src/utils/image.js

```diff
@@ -37,3 +43,40 @@ export async function getThumbnailData(filePromise) {
         return { filename: file?.name ?? '', thumbnail: '' };
     }
 }
+
+/**
+ * Utility hook for handling file values in file upload fields.
+ * This custom hook maintains a referentially-stable list of files,
+ * and will execute a callback every time that list changes.
+ */
+export function useFileData(initialFiles, callback) {
```

Comment:
can we default callback to () => {} so we don't have to check if callback exists before running it